### PR TITLE
Update GPT2 models for OSS

### PR DESCRIPTION
**Pull request by novaturient95**
### GPT2 
**run to save model to blobstorage:** [link](https://ml.azure.com/experiments/id/8909346e-1431-4ab9-9a4f-f79d92ee10ae/runs/sad_skin_mjbgtvj42c?wsid=/subscriptions/80c77c76-74ba-4c8c-8229-4c3b2957990c/resourceGroups/huggingface-registry-test1/providers/Microsoft.MachineLearningServices/workspaces/test-eastus&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#)

**Batch validation run:** [link](https://ml.azure.com/runs/batchjob-460a3cfa-fbc4-e0dc-dc1e-6f37e059ec17?wsid=/subscriptions/80c77c76-74ba-4c8c-8229-4c3b2957990c/resourceGroups/huggingface-registry-test1/providers/Microsoft.MachineLearningServices/workspaces/test-eastus&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#/?graphId=0c909a9e-509a-4f13-be63-d435fe6045f0&label=plum_foot_y9p1tdzl&path=%2Fruns%2Fbatchjob-460a3cfa-fbc4-e0dc-dc1e-6f37e059ec17&runId=batchjob-460a3cfa-fbc4-e0dc-dc1e-6f37e059ec17)

**Online inference validation** 
endpoint: [link](https://ml.azure.com/endpoints/realtime/gpt-oss-retest-di/test?wsid=/subscriptions/80c77c76-74ba-4c8c-8229-4c3b2957990c/resourceGroups/huggingface-registry-test1/providers/Microsoft.MachineLearningServices/workspaces/test-eastus&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)

image

**MLModel**

```
flavors:
  python_function:
    env:
      conda: conda.yaml
      virtualenv: python_env.yaml
    loader_module: mlflow.transformers
    model_binary: model
    python_version: 3.8.18
  transformers:
    code: null
    components:
    - tokenizer
    framework: pt
    instance_type: TextGenerationPipeline
    model_binary: model
    pipeline_model_type: GPT2LMHeadModel
    task: text-generation
    tokenizer_type: GPT2TokenizerFast
    transformers_version: 4.35.2
metadata:
  base_model_name: gpt2
  base_model_task: text-generation
  is_acft_model: true
  is_finetuned_model: false
mlflow_version: 2.8.1
model_size_bytes: 501138122
model_uuid: c49bee2e635f40a6a62b38ede9b0e6c8
signature:
  inputs: '[{"type": "string"}]'
  outputs: '[{"type": "string"}]'
  params: '[{"name": "top_p", "type": "float", "default": 1.0, "shape": null}, {"name": "temperature", "type": "float", "default": 1.0, "shape": null}, {"name": "max_new_tokens", "type": "integer", "default": 50, "shape": null}, {"name": "do_sample", "type": "boolean", "default": true, "shape": null}, {"name": "return_full_text", "type": "boolean", "default": true, "shape": null}]'
utc_time_created: '2023-12-06 11:43:15.922970'

```

**conda.yaml**

```
channels:
- anaconda
- conda-forge
dependencies:
- python=3.8.18
- pip:
  - accelerate==0.23.0
  - cffi==1.16.0
  - dill==0.3.7
  - google-api-core==2.12.0
  - mlflow==2.8.1
  - numpy==1.24.4
  - packaging==23.0
  - pillow==10.0.1
  - protobuf==3.20.3
  - pyyaml==6.0.1
  - requests==2.31.0
  - scikit-learn==1.3.2
  - scipy==1.10.1
  - sentencepiece==0.1.99
  - torch==1.13.1
  - torchvision==0.14.1
  - transformers==4.34.1
- pycocotools=2.0.4
name: project_environment

```

## GPT2-Large

**run to save model to blobstorage:** [run](https://ml.azure.com/experiments/id/8909346e-1431-4ab9-9a4f-f79d92ee10ae/runs/blue_cup_68mhfn77zh?wsid=/subscriptions/80c77c76-74ba-4c8c-8229-4c3b2957990c/resourceGroups/huggingface-registry-test1/providers/Microsoft.MachineLearningServices/workspaces/test-eastus&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#)

**batch validation run**: [run](https://ml.azure.com/runs/batchjob-78c2b4aa-cce5-6bcb-099e-b6266ebe3afb?wsid=/subscriptions/80c77c76-74ba-4c8c-8229-4c3b2957990c/resourceGroups/huggingface-registry-test1/providers/Microsoft.MachineLearningServices/workspaces/test-eastus&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#)

**online validation run**

**MLmodel:**
```
flavors:
  python_function:
    env:
      conda: conda.yaml
      virtualenv: python_env.yaml
    loader_module: mlflow.transformers
    model_binary: model
    python_version: 3.8.18
  transformers:
    code: null
    components:
    - tokenizer
    framework: pt
    instance_type: TextGenerationPipeline
    model_binary: model
    pipeline_model_type: GPT2LMHeadModel
    task: text-generation
    tokenizer_type: GPT2TokenizerFast
    transformers_version: 4.35.2
metadata:
  base_model_name: gpt2-large
  base_model_task: text-generation
  is_acft_model: true
  is_finetuned_model: false
mlflow_version: 2.8.1
model_size_bytes: 3099561926
model_uuid: 916fcb500dd44c7c84eb43ec02107e05
signature:
  inputs: '[{"type": "string"}]'
  outputs: '[{"type": "string"}]'
  params: '[{"name": "top_p", "type": "float", "default": 1.0, "shape": null}, {"name": "temperature", "type": "float", "default": 1.0, "shape": null}, {"name": "max_new_tokens", "type": "integer", "default": 50, "shape": null}, {"name": "do_sample", "type": "boolean", "default": true, "shape": null}, {"name": "return_full_text", "type": "boolean", "default": true, "shape": null}]'
utc_time_created: '2023-12-06 09:18:29.934345'

```

**conda.yaml**

```
channels:
- anaconda
- conda-forge
dependencies:
- python=3.8.18
- pip:
  - accelerate==0.23.0
  - cffi==1.16.0
  - dill==0.3.7
  - google-api-core==2.12.0
  - mlflow==2.8.1
  - numpy==1.24.4
  - packaging==23.0
  - pillow==10.0.1
  - protobuf==3.20.3
  - pyyaml==6.0.1
  - requests==2.31.0
  - scikit-learn==1.3.2
  - scipy==1.10.1
  - sentencepiece==0.1.99
  - torch==1.13.1
  - torchvision==0.14.1
  - transformers==4.34.1
- pycocotools=2.0.4
name: project_environment
```

## GPT2 Medium

**run to save model to blobstorage:** [link](https://ml.azure.com/experiments/id/8909346e-1431-4ab9-9a4f-f79d92ee10ae/runs/good_yacht_kkrw5f1jk0?wsid=%2Fsubscriptions%2F80c77c76-74ba-4c8c-8229-4c3b2957990c%2FresourceGroups%2Fhuggingface-registry-test1%2Fproviders%2FMicrosoft.MachineLearningServices%2Fworkspaces%2Ftest-eastus&tid=72f988bf-86f1-41af-91ab-2d7cd011db47&reloadCount=1#)

**batch validation:** [link](https://ml.azure.com/runs/batchjob-57f1d978-bc68-b5db-f159-3c94d5723023?wsid=/subscriptions/80c77c76-74ba-4c8c-8229-4c3b2957990c/resourceGroups/huggingface-registry-test1/providers/Microsoft.MachineLearningServices/workspaces/test-eastus&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#/?graphId=63b23ca0-4479-4547-9907-059422bba29c&label=quirky_oven_f5mdszxk&path=%2Fruns%2Fbatchjob-57f1d978-bc68-b5db-f159-3c94d5723023&runId=batchjob-57f1d978-bc68-b5db-f159-3c94d5723023)

**MLmodel**
```
flavors:
  python_function:
    env:
      conda: conda.yaml
      virtualenv: python_env.yaml
    loader_module: mlflow.transformers
    model_binary: model
    python_version: 3.8.18
  transformers:
    code: null
    components:
    - tokenizer
    framework: pt
    instance_type: TextGenerationPipeline
    model_binary: model
    pipeline_model_type: GPT2LMHeadModel
    task: text-generation
    tokenizer_type: GPT2TokenizerFast
    transformers_version: 4.35.2
metadata:
  base_model_name: gpt2-medium
  base_model_task: text-generation
  is_acft_model: true
  is_finetuned_model: false
mlflow_version: 2.8.1
model_size_bytes: 1422708538
model_uuid: 798d0a704f874098b395bbe1fdcaa86c
signature:
  inputs: '[{"type": "string"}]'
  outputs: '[{"type": "string"}]'
  params: '[{"name": "top_p", "type": "float", "default": 1.0, "shape": null}, {"name": "temperature", "type": "float", "default": 1.0, "shape": null}, {"name": "max_new_tokens", "type": "integer", "default": 50, "shape": null}, {"name": "do_sample", "type": "boolean", "default": true, "shape": null}, {"name": "return_full_text", "type": "boolean", "default": true, "shape": null}]'
utc_time_created: '2023-12-06 09:45:45.494930'

```

**conda.yaml**

```
channels:
- anaconda
- conda-forge
dependencies:
- python=3.8.18
- pip:
  - accelerate==0.23.0
  - cffi==1.16.0
  - dill==0.3.7
  - google-api-core==2.12.0
  - mlflow==2.8.1
  - numpy==1.24.4
  - packaging==23.0
  - pillow==10.0.1
  - protobuf==3.20.3
  - pyyaml==6.0.1
  - requests==2.31.0
  - scikit-learn==1.3.2
  - scipy==1.10.1
  - sentencepiece==0.1.99
  - torch==1.13.1
  - torchvision==0.14.1
  - transformers==4.34.1
- pycocotools=2.0.4
name: project_environment

```

### CODE REVIEW

File: assets/models/system/gpt2/description.md [hunk 1]
```diff
@@ -1,50 +1,90 @@
...
+GPT-2 is a transformers model pretrained on a very large corpus of English data in a self-supervised fashion. This means it was pretrained on the raw texts only, with no humans labelling them in any way (which is why it can use lots of publicly available data) with an automatic process to generate inputs and labels from those texts. More precisely, it was trained to guess the next word in sentences.
 
+This is the smallest version of GPT-2, with 124M parameters.
 
-> The above summary was generated using ChatGPT. Review the <a href="https://huggingface.co/gpt2" target="_blank">original model card</a> to understand the data used to train the model, evaluation metrics, license, intended uses, limitations and bias before using the model.
+**Resources for more information:**
```
Comment (novaturient95) [inline assets/models/system/gpt2/description.md:5]: not present in HF model card for gpt2 but repeating pattern for gpt2-medium and large and hence included

File: assets/models/system/gpt2/description.md [hunk 2]
```diff
@@ -1,50 +1,90 @@
...
-    },
-    {
-        "0": "Once upon a time, they were just another small family, only three. She says one day that her father was getting a new license"
-    }
+  "Once upon a time, far far away, of this world he lived. As soon as his son's dying in a car explosion, he came to an awareness that this world was far from the real thing. He was aware of the life that he had left behind him.\n\nThis awareness",
+  "Long time ago, there was a man who took that question away from me: not the first case where someone called me. He had called me, but he was so upset and depressed about it from the bottom of his heart, he was crying, and he thought my name was Kimber"
 ]
 ```
+
+## Limitations and bias
```
Comment (novaturient95) [inline assets/models/system/gpt2/description.md:75]: Added Limitation and bias at the bottom as discussed

File: assets/models/system/gpt2-medium/description.md [hunk 3]
```diff
@@ -1,49 +1,112 @@
...
+Batch |<a href="https://aka.ms/azureml-infer-batch-sdk-text-generation" target="_blank">text-generation-batch-endpoint.ipynb</a>| coming soon
+
+## Sample inputs and outputs (for real-time inference)
+
+### Sample input
 ```json
 {
-    "input_data": {
-        "input_string": ["My name is John and I am", "Once upon a time,"]
+    "input_data": [
```
Comment (novaturient95) [inline assets/models/system/gpt2-medium/description.md:83]: samples here are from gpt2 endpoint. Model need to be re-deployed and checked

File: assets/models/system/gpt2-large/description.md [hunk 4]
```diff
@@ -1,50 +1,112 @@
...
+
+
+## Sample inputs and outputs (for real-time inference)
+
+### Sample input
 ```json
 {
-    "input_data": {
-        "input_string": ["My name is John and I am", "Once upon a time,"]
+    "input_data": [
```
Comment (novaturient95) [inline assets/models/system/gpt2-large/description.md:84]: Samples here are from gpt2 endpoint. Model need to be re-deployed and checked

**Review by tinaem — APPROVED** [review]

**Review by BhavanaTumma — APPROVED** [review]
Approved.

**Pull request merged**